### PR TITLE
Grep enhancements

### PR DESCRIPTION
**Pull request by funktioniert**
Grep (~) is now able to print specific columns. E.g.:
```bash
⚔ r2 malloc://1024
[0x00000000]> wx 102030405060708090
[0x00000000]> px~:1[3,5,0-1]
0x00000000 1020 5060 9000
```
Apart from that i updated the usage message to include more info of already available features like '$' and cleaned up libr/cons/grep.c to conform to the coding guidelines.

### CODE REVIEW

**Comment by funktioniert** [comment]
Sorry, missed those checks in radare2-regression. Fixing now.

File: libr/cons/grep.c [hunk 1]
```diff
@@ -46,20 +54,19 @@ R_API void r_cons_grep(const char *str) {
 	cons->grep.begin = 0;
 	cons->grep.counter = 0;
 	cons->grep.nstrings = 0;
-	cons->grep.tokenfrom = 0;
-	cons->grep.tokento = ST32_MAX;
+	memset (cons->grep.tokens, 0, R_CONS_GREP_TOKENS);
```
Comment (jvoisin) [inline libr/cons/grep.c]: What about this instead:
```
memset (cons->grep, 0, sizeof (cons->grep));
cons->grep.line = -1;
```

Comment (funktioniert) [inline libr/cons/grep.c]: Oh nice. Thanks.

**Comment by radare** [comment]
What feature are you trying to implemwnt here? Because current grep already supporting getting speicifc column or row

**Comment by radare** [comment]
Ok i see, that it supports a list with items and ranges. Good!

**Comment by funktioniert** [comment]
Exactly. Sorry for not being specific enough.

**Comment by radare** [comment]
Can you squash all those commits into a single one? (git rebase -i @~9), mark all except the first as 's', edit the commit message and push -f

**Comment by radare** [comment]
Nevermind, i did it on my side. lgtm, merged :)

**Comment by funktioniert** [comment]
Sorry to bother, but I think u might have forgotten to merge or push?

**Comment by radare** [comment]
its already merged and pushed. just squashed and cherry-picked.

btw, looks like the `?@?` help message doesnt mentions that new syntax. can you send another PR updating it? also, it would be good to have more tests, checking more ranges, and negated conditions or out of range items.

thanks!

**Comment by radare** [comment]
it's here https://github.com/radare/radare2/commit/7bb5f394956ed2cddc62f7371b55c6f80f4b3141
